x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_embeddable_factory.tsx

```diff
@@ -211,11 +212,13 @@ export const getAnomalySwimLaneEmbeddableFactory = (
           const I18nContext = i18n.Context;
           const timeBuckets = useTimeBuckets(uiSettings);
 
+          if (!apiHasExecutionContext(parentApi)) {
+            throw new Error('Parent API does not have execution context');
+          }
```

Comment:
Is it possible to move this check to the top of the `buildEmbeddable` method rather than inside the component?

Reply:
I also wonder if I should emit with the `blockingError` subject?

Reply:
@nickpeihl I assume we need it at both the API and component level because we still can render embeddable with `ReactEmbeddableRenderer` and provide the `parentApi` prop.

Reply:
> I also wonder if I should emit with the blockingError subject?

This is in line with what I was thinking.

> @nickpeihl I assume we need it at both the API and component level because we still can render embeddable with ReactEmbeddableRenderer and provide the parentApi prop.

Ah, I understand now. I had forgotten you were using `ReactEmbeddableRender` to render the component elsewhere in your plugin. What does the component render when the error is thrown?

Reply:
nit: I'm still not understanding why this error needs to be thrown in both the component and in the top of the `buildEmbeddable` method? 

IMO it should be only in `buildEmbeddable`. Is there any way to render this component _without_ the `ReactEmbeddableRender`?

Reply:
It was the quickest way to migrate usage of [AnomalySwimLane](https://github.com/elastic/kibana/blob/2f14fb8d527c8b2823a2c475ccfc08eeb8186e6b/x-pack/plugins/ml/public/shared_components/anomaly_swim_lane.tsx#L26)  in the Logs UI, but potentially - yes, we can remove `ReactEmbeddableRender` with some extra work